modules/islandora_entities_csv_import/includes/islandora_entities_csv_import.form.inc

```diff
@@ -57,3 +58,42 @@ function islandora_entities_import_form_submit(array $form, array &$form_state)
   $collection_pid = $form_state['values']['collection'];
   islandora_entities_import_csv_batch(drupal_realpath($file->uri), $collection_pid);
 }
+
+/**
+ * Validates the collection Drupal Form Element.
+ *
+ * Ensures that the object exists, is a collection and has the correct policy.
+ *
+ * @param array $element
+ *   The collection Drupal Form Element.
+ * @param array $form_state
+ *   The Drupal Form State.
+ */
+function islandora_entities_csv_import_validate_collection(array $element, array &$form_state) {
+  $collection_pid = &$form_state['values']['collection'];
+  $collection_object = islandora_object_load($collection_pid);
+  if ($collection_object) {
```

Comment:
From line 75, most of the code looks pretty similar to the one in each individual batch process. Seems to me we could make a single "check if collection exists and policy matches what we are passing" function and reuse it here and in batch handler down here https://github.com/Islandora/islandora_solution_pack_entities/pull/125/files#diff-100a7d9d528fcbde5b9c60729bd2d173R76? ?

Reply:
You are right, one could pull out some re-used code into a custom function...